File: scm-csharp/scheme/SchemeWebSocket.cs
using System.Net.Sockets;

namespace scheme;

public class SchemeWebSocket
{
    public NetworkStream stream;
    public bool IsServer;

    public SchemeWebSocket(NetworkStream stream, bool isServer)
    {
        this.stream = stream;
        this.IsServer = isServer;
    }

    public override string ToString() => "#<websocket>";
}

File: scm-java/src/main/scheme/SchemeWebSocket.java
package scheme;

import java.io.InputStream;
import java.io.OutputStream;

public class SchemeWebSocket {
    public InputStream input;
    public OutputStream output;
    public boolean isServer;

    public SchemeWebSocket(InputStream input, OutputStream output, boolean isServer) {
        this.input = input;
        this.output = output;
        this.isServer = isServer;
    }

    @Override
    public String toString() { return "#<websocket>"; }
}
